app/services/translation_verificator.rb

```diff
@@ -0,0 +1,47 @@
+class TranslationVerificator
+  def initialize(card, user_translation, response_time)
+    @response_time = response_time
+    @user_translation = user_translation
+    @card = card
+  end
+
+  def result
+    { number_of_reviews: calculate_number_of_reviews,
+      review_date: calculate_review_date,
+      e_factor: calculate_e_factor,
+      interval: calculate_interval }
+  end
+
+  def assess_quality
+    QualityAssessor.new(@response_time,
+                        @user_translation,
+                        @card.original_text).result
+  end
+
+  private
+
+  def calculate_number_of_reviews
+    case assess_quality
+    when 3..5 then @card.number_of_reviews += 1
+    else @card.number_of_reviews = 0
+    end
+  end
+
+  def calculate_review_date
```

Comment:
Assignment Branch Condition size for calculate_review_date is too high. [15.3/15]